### DISCUSSION TITLE
Jellyfin access to nfs media folder on NAS

### DISCUSSION THREAD

**Discussion by collowe** [tteck/Proxmox#2038] [Help Needed]
Hi

I've recently moved from a docker instance of Jellyfin to an LXC in Proxmox using the vrey helpful scripts here (thanks!).  I cannot seem to get Jellyfin to access my shared media folder on my NAS which is mapped to the LXC as an NFS folder:

mp0: /mnt/pve/movies,mp=/mnt/movies

My movies folder has the following owner setup, as shown from within the jellyfin lxc:

root@jellyfin:/mnt/movies# ls -al
total 20947512
drwxrwxrwx 11 100000 100000        4096 Nov 12 07:52  .
drwxr-xr-x  3 root   root          4096 May  4  2023  ..
drwxrwxrwx  2 100000 100000        4096 Oct 19  2022  .@__thumb
drwxrwxrwx  2 100000 100000        4096 Aug 13 22:21 'movie1
drwxrwxrwx  2 100000 100000        4096 Aug 12 00:53 'movie2'
drwxrwxrwx  2 100000 100000        4096 Aug 12 01:34 'movie3'

This setup works with the arr apps when I complete idmapping in the lxc config as per:

lxc.idmap: u 0 100000 1000
lxc.idmap: g 0 100000 1000
lxc.idmap: u 1000 1000 1
lxc.idmap: g 1000 1000 1
lxc.idmap: u 1001 101000 64535
lxc.idmap: g 1001 101000 64535

However this doesn't work in the jellyfin lxc - there are permissions issues.  Digging into this, jellyfin runs as user 110 in group 118 in the lxc: 

root@jellyfin:/var/log/jellyfin# less /etc/passwd | grep jellyfin
jellyfin:x:110:118:Jellyfin default user,,,:/var/lib/jellyfin:/bin/false

So I tried mapping 110 to 100000 and root to root: 

lxc.idmap: u 0 0 110
lxc.idmap: g 0 0 118
lxc.idmap: u 110 100000 1
lxc.idmap: g 118 100000 1
lxc.idmap: u 111 111 889
lxc.idmap: g 119 119 881
lxc.idmap: u 1000 1000 1
lxc.idmap: g 1000 1000 1
lxc.idmap: u 1001 101000 64535
lxc.idmap: g 1001 101000 64535

which seems to give jellyfin access to the files:

root@jellyfin:/mnt/movies# ls -al
total 20947512
drwxrwxrwx 11 jellyfin jellyfin        4096 Nov 12 07:52  .
drwxr-xr-x  3 root     root            4096 May  4  2023  ..
drwxrwxrwx  2 jellyfin jellyfin        4096 Oct 19  2022  .@__thumb
drwxrwxrwx  2 jellyfin jellyfin        4096 Aug 13 22:21 'movie1'
drwxrwxrwx  2 jellyfin jellyfin        4096 Aug 12 00:53 'movie2'

However the jellyfin UI now has stopped working and gives and unable to connect message.  I also get permissions errors viewing the error log in /var/logs/jellyfin.

I've seen similar discussions so hoping someone has cracked this issue or has an alternative approach.  I cannot see anything relating to giving the jellyfin user the necessary permissions outside the lxc though.  Ideally I wouldn't have to rework permissions for all my arr apps and can make Jellyfin work with this existing setup.  Any thoughts welcome.  I'm not that familiar with idmapping so its entirely possible I haven't got that quite right.

**Comment by onethree7** [comment]
Is this an unpriviledged container or a priviledged one? 
Whenever i am sharing folders in whatever protocol i found i am way more successfull if my containers are priviledged. 

In case you want to switch this container to priviledged you can either redeploy or backup and re-import it as privil.. , but its just a rather unedacutated guess from myself , i get with the three keywords "user rights, container, UI stopped working"...

**Reply by collowe** [reply]
It's an unprivileged container, appreciate it is easier with a privileged flavour.

May well switch if I cannot get to an answer with unpriv.  It feels like it should be doable though.

**Reply by onethree7** [reply]
if it is unpriv, I think adding the mount to the PVE and from there to the container as a directory or so would work. I am not sure though. Can unpriviledged even do HW transcoding?

**Reply by collowe** [reply]
Actually, I've just realised that it is privileged thanks to your question...  I'd just presumed it was unpriv like the others....  doh.  So now looking at why I get a permissions error on the NAS via the mount.

**Reply by yodaphone** [reply]
you can use unprivileged container & use bind points or preferably with privileged container. emable CIFS, SMB in Option for the container

**Reply by tteck** [reply]
If you use unprivileged, you'll have to bring your own "Hardware Acceleration Support"

**Comment by yodaphone** [comment]
this is very easy to do

1. Mount the NFS Share in Proxmox Host / Create Storage under Datacenter (You can use the UI and say it contains snippets. it really doesnt matter)

you cannot map the root folder in a NAS so you may want to map a folder like for example "music"

3. create a directory inside the container 

mkdir -p /mnt/nasmedia

4. On the host run the following command

pct set 100(whatever your container id is) -mp0 /mnt/pve/music,mp=/mnt/nasmedia

thats it. not reboot the container and you will be able to see the folder inside the container

**Reply by lordratner** [reply]
This is the answer. Here's some more detail if needed:

https://forum.proxmox.com/threads/jellyfin-lxc-with-nvidia-gpu-transcoding-and-network-storage.138873/
